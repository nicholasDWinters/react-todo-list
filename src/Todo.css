.Todo {
    margin-top: 20px;
    margin-bottom: 20px;
}

.Todo-task {
    margin-right: 10px;
}