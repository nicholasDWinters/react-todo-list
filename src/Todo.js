import React, { useState } from 'react';
import './Todo.css';
import EditTodoForm from './EditTodoForm';

const Todo = ({ task, id, removeTodo, editTodo }) => {
    const todo = {
        task,
        id
    };
    const [hidden, setHidden] = useState(true);
    const resetHidden = () => {
        setHidden(true);
    }
    const rendered = hidden ? { display: 'none' } : {};
    return (
        <div className='Todo'>
            <span className='Todo-task'>{task}</span>
            <button onClick={() => setHidden(false)}>Edit</button>
            <button onClick={() => removeTodo(id)}>X</button>
            <EditTodoForm editTodo={editTodo} todo={todo} resetHidden={resetHidden} rendered={rendered} />
        </div>
    )
}

export default Todo;