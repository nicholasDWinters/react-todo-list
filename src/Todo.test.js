import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Todo from './Todo';

it("renders without crashing", function () {
    render(<Todo />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it("should display edit form on click", function () {
    const { queryByText, getByLabelText } = render(<Todo />);
    const btn = screen.getByRole('button', { name: 'Edit' });
    fireEvent.click(btn);
    const input = getByLabelText("Edit");
    const editBtn = queryByText("Edit Todo");
    expect(input).toBeInTheDocument();
    expect(editBtn).toBeInTheDocument();
})
