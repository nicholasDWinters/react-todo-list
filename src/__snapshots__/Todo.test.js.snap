// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span
      class="Todo-task"
    />
    <button>
      Edit
    </button>
    <button>
      X
    </button>
    <form
      class="EditTodoForm"
      style="display: none;"
    >
      <label
        class="EditTodoForm-lable"
        for="todo"
      >
        Edit
      </label>
      <input
        class="EditTodoForm-input"
        id="todo"
        name="task"
        placeholder="Edit Todo"
        type="text"
        value=""
      />
      <button
        class="EditTodoForm-button"
      >
        Edit Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
